cli/session.go

```diff
@@ -0,0 +1,26 @@
+package cli
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/coder/coder/cli/cliui"
+)
+
+func session() *cobra.Command {
```

Comment:
I think this going under a config `config` would scale better. We currently have a few options that are saved:
- Session token
- Deployment URL
- Organization (not used for single tenant, but will be once multiple are added)
- Dotfiles URL
- Built-in PostgreSQL path, password, and port

Something like:

```
$ coder config get session
<token prints>
```

could be nice.

Reply:
That's an interesting distinction that I buy. How would one go about printing the currently authenticated organization then?

Reply:
I think the other use case should use perhaps a `deployment` or `admin` subcommand.  `config` may not sufficiently clearly distinguish between ordinary user/local and privileged user/deployment.

Reply:
I'd like another opinion on this naming, as I've typically seen this under a `config` command, but the rationale makes a lot of sense here. cc: @mafredri @johnstcn any thoughts?

Reply:
> How would one go about printing the currently authenticated organization then?

If you mean multitenancy, I think this should print all active sessions.  Otherwise I'd go with what's outlined in #2786 .

Reply:
Some thoughts:
- `coder env` could also work (think `go env`) -- it also feels like it could be output in a sourceable format
- Agree regarding `deployment`/`admin`/ subcommand, I'd feel strange about putting the deployment-related variable and user-related variables under the same subcommand

Reply:
I think there's a case to be made for `coder config`, but I see it as something down-the-line requiring a re-think, some thoughts related to that:

- Config would be one place to get and set anything related to everything
- There could be a `coder config show` for an overview of everything
- How do we want configuration to work in the future? It'd be pretty neat to be able to define instances/deployments in a `yaml` file, and change settings per-instance (e.g. instance X has `CODER_SSH_FORWARD_AGENT=true`, instance Y has `CODER_SSH_FORWARD_AGENT=false`)
- Personally, I don't find the `config` and `config-ssh` collision to be that bad, although `config-ssh` could be renamed `configure-ssh` (with an alias) in this case
- Pro for `get session` vs no `get`, allows for programmatic use, if the user knows what they want they can use `mytoken=$(coder config get session_token)` (or add a `--instance Y` for a specific instance)
  - As output in this PR, deciding how we'd allow programmatic use becomes harder, especially when outputting multiple values

That said, I also don't think `coder session` and `coder config` need to be mutually exclusive, it'd be OK to have both. I also like @johnstcn's idea about `coder env` which would be a nice feature addition in general (not necessarily dependent on the decision here).

Reply:
> Config would be one place to get and set anything related to everything

I dunno, I think it's pretty important to distinguish between client configuration/state and server configuration/state.

Being able to extract just a single field seems like a good use case, but I'd rather either use a `--` flag (like `uname --operating-system` (which has the short `-o`) *or* have a structured text output like JSON where they can use `jq` to pluck out arbitrary fields.  `get` is a pretty vague word.  Git uses it for generic key/value configuration settings and uses `git config --get <key>` not `git get <key>`.

Reply:
> I dunno, I think it's pretty important to distinguish between client configuration/state and server configuration/state.

I agree with this in principle, but I see the clients current session token more as user configuration than server. But maybe I'm not thinking about it the right way.

A bastardized example out of `kubectl` config (`~/.kube/config`) would be:

```yaml
apiVersion: v1
clusters:
- cluster:
    server: https://dev.coder.com
  name: codercom
contexts:
- context:
    cluster: codercom
    user: dev
  name: codercom
current-context: codercom
kind: Config
preferences: {}
users:
- name: dev
  user:
    token: session-token-95B8A61C-ECAF-43F7-8958-3ED769AEF953
```

I'm not saying this is at all what our configuration would look like, but the essence of instances and users are there (including tokens).

Regarding flags (e.g. `--session-token`) I'm not opposed as all, I think that makes sense as long as there aren't many (say, more than 5) things that can be output (which would be the case here). If we'd go for something like `config [--]get`, I think flag or non-flag would both be fine options. As you say, `git` does `--get`, whereas e.g. `gh` does `get`. `kubectl` only seems to do ~`set` and `view`, where `view` outputs json, also a fine option IMO.